internal/app/app.go

```diff
@@ -165,11 +167,19 @@ func (app *App) RunNonInteractive(ctx context.Context, prompt string, quiet bool
 	messageEvents := app.Messages.Subscribe(ctx)
 	messageReadBytes := make(map[string]int)
 
-	defer fmt.Printf(ansi.ResetProgressBar)
+	defer func() {
+		_, _ = fmt.Printf(ansi.ResetProgressBar)
+
+		// Always print a newline at the end. If output is a TTY this will
+		// prevent the prompt from overwriting the last line of output.
+		_, _ = fmt.Fprintln(output)
+	}()
+
 	for {
-		// HACK: add it again on every iteration so it doesn't get hidden by
-		// the terminal due to inactivity.
-		fmt.Printf(ansi.SetIndeterminateProgressBar)
+		// HACK: Reinitialize the terminal progress bar on every iteration so
+		// it doesn't get hidden by the terminal due to inactivity.
+		_, _ = fmt.Printf(ansi.SetIndeterminateProgressBar)
```

Comment:
Does this cause the some bugs on iTerm2/SSH that https://github.com/charmbracelet/crush/pull/1329 fixed for interactive?

If so, we might want to address them.

/cc @aymanbagabas